### ISSUE TITLE
[FEATURE] Expose convenient interfaces by agent

### ISSUE THREAD

**Issue by priyanshunayan** [HTTP-APIs/hydra-python-agent#150]
### I'm submitting a 
- [ ] bug report.
- [x] feature request.

### Current Behaviour:

Presently Agent provides only CRUD operations for developers to interact with the APIDOC. It would be helpful to expose different methods like getSupportedOperations or getMembers. These methods can allow developers to take actions programatically, instead of parsing the response themselves. It would be nice to have methods similar as reference implementation: https://github.com/HydraCG/Heracles.ts/tree/master/docs#functions. Also, right now response is returned in the form of JSON, it would be nice have it transformed into dict so that user can reference members of response easily. 
### Expected Behaviour:

One would be able to query from the dict like they normally do. for eg: `res.members` to get members of the collections. 

### Steps to reproduce:

NA

### Do you want to work on this issue?

yes

**Comment by xadahiya** [comment]
Not a bad idea! @chrizandr what do you think?

**Comment by chrizandr** [comment]
Yes this seems useful. Heracles already parses the response of the API, we should try something similar. I don't think getSupportedOperations would be a part of the response object, we already have supported operations in core library when the API Doc is parsed into the HydraDoc object. 

It would be a good idea to consolidate how responses can be stored as python objects and how we parse the API responses, this would be a great addition to the core library and then we could use core in agent.
